zebra-test/src/network_addr.rs

```diff
@@ -0,0 +1,303 @@
+                "0100000000000000",                 // service flags, NODE_NETWORK
+                "00000000000000000000ffff",         // address, fixed 16 bytes (IPv4 embedded in IPv6)
+                "7f000001",                         // IPv4, 127.0.0.1
+                "0000",                             // port
+            )
+        ).expect("Message bytes are in valid hex representation"),
+
+    ];
+
+    /// Array of empty or unsupported `addr` (v1) test vectors.
+    ///
+    /// These test vectors can be read by [`zebra_network::protocol::external::Codec::read_addr`].
+    /// They should produce successful but empty results.
+    pub static ref ADDR_V1_EMPTY_VECTORS: Vec<Vec<u8>> = vec![
+        // Empty list
+        <Vec<u8>>::from_hex(
+            "00" // number of entries
+        ).expect("Message bytes are in valid hex representation"),
+    ];
+
+    /// Array of ZIP-155 test vectors containing IP addresses.
+    ///
+    /// Some test vectors also contain some unsupported addresses.
+    ///
+    /// These test vectors can be read by [`zebra_network::protocol::external::Codec::read_addrv2`],
+    /// They should produce successful results containing IP addresses.
+    // From https://github.com/zingolabs/zcash/blob/9ee66e423a3fbf4829ffeec354e82f4fbceff864/src/test/netbase_tests.cpp#L421
+    pub static ref ADDR_V2_IP_VECTORS: Vec<Vec<u8>> = vec![
+        // stream_addrv2_hex
+        <Vec<u8>>::from_hex(
+            concat!(
+                "03", // number of entries
+
+                "61bc6649",                         // time, Fri Jan  9 02:54:25 UTC 2009
+                "00",                               // service flags, COMPACTSIZE(NODE_NONE)
+                "02",                               // network id, IPv6
+                "10",                               // address length, COMPACTSIZE(16)
+                "00000000000000000000000000000001", // address
+                "0000",                             // port
+
+                "79627683",                         // time, Tue Nov 22 11:22:33 UTC 2039
+                "01",                               // service flags, COMPACTSIZE(NODE_NETWORK)
+                "02",                               // network id, IPv6
+                "10",                               // address length, COMPACTSIZE(16)
+                "00000000000000000000000000000001", // address
+                "00f1",                             // port
+
+                "79627683",                         // time, Tue Nov 22 11:22:33 UTC 2039
+                "01",                               // service flags, COMPACTSIZE(NODE_NETWORK)
+                "04",                               // network id, TorV3
+                "20",                               // address length, COMPACTSIZE(32)
+                "53cd5648488c4707914182655b7664034e09e66f7e8cbf1084e654eb56c5bd88", // 32 bytes of key + 2 bytes checksum
```

Comment:
No, the checksum only exists to avoid mistakes in human address entry. It is deterministically calculated from the public key, so it doesn't need to be sent by machines.

(I helped design this format!)

Fixed in a7f11c210

Reply:
> (I helped design this format!)

That's awesome! :star_struck: